pandas_datareader/tests/yahoo/test_yahoo.py

```diff
@@ -131,6 +131,19 @@ def test_get_data_multiple_symbols(self):
         sl = ['AAPL', 'AMZN', 'GOOG']
         web.get_data_yahoo(sl, '2012')
 
+    @pytest.mark.parametrize('adj_pr', [True, False])
+    def test_get_data_null_as_missing_data(self, adj_pr):
+        # workaround as skip_on_exception decorator changes signature
+        # should it be changed to a signature-preserving decorator?
```

Comment:
Decorators generally are difficult to make signature-preserving AFAIK.  Does `pytest` and this decorator not work well together?  Did you try using `skip_on_exception` above the `pytest.mark` one?